commit 9cf04af7d152c1dfc7e74342b530155a7697cd3e (refs/changes/12/5312/5)
Author: Robert Bogart <Robert.bogart@joyent.com>
Date:   2019-01-18T22:34:55+00:00 (9 months ago)
    
    MANATEE-343 manatee full rebuilds should pull from the primary where possible
    Reviewed by: Richard Bradley <richard.bradley@joyent.com>
    Approved by: Joshua M. Clulow <jmc@joyent.com>

diff --git a/lib/postgresMgr.js b/lib/postgresMgr.js
index 5ac122f..8929ed2 100644
--- a/lib/postgresMgr.js
+++ b/lib/postgresMgr.js
@@ -934,8 +934,19 @@ PostgresMgr.prototype._reconfigure = function _reconfigure(pgConfig, callback) {
         var pgUrl = pgConfig.downstream ? pgConfig.downstream.pgUrl : null;
         return (self._primary(pgUrl, onReconfigure));
     } else {
+        /*
+         * If this peer is sync or async, then use `backupUrl' supplied in the
+         * object `restorePeer`.  All peers in the cluster perform rebuilds off
+         * of the primary as a way of applying back pressure to it so that the
+         * sync will not fall hopelessly behind when it is under heavy load.
+         * For postgres-based replication, we should continue to use whoever our
+         * upstream-peer would be -- in the case of the sync, that will be the
+         * primary and in the case of an async, the upstream peer will either
+         * be the sync or another async depending on how many peers there are in
+         * the cluster.
+         */
         return (self._standby(pgConfig.upstream.pgUrl,
-                              pgConfig.upstream.backupUrl,
+                              pgConfig.restorePeer.backupUrl,
                               onReconfigure));
     }
 };
